src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -0,0 +1,82 @@
+﻿using Diagnostics.DataProviders;
+using Diagnostics.RuntimeHost.Models;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Diagnostics.ModelsAndUtils.Models.Storage;
+using Newtonsoft.Json;
+using Diagnostics.RuntimeHost.Utilities;
+using System.IO;
+using Diagnostics.Logger;
+
+namespace Diagnostics.RuntimeHost.Services.SourceWatcher.Watchers
+{
+    public class StorageWatcher : ISourceWatcher
+    {
+        private IStorageService storageService;
+        private IHostingEnvironment hostingEnvironment;
+        private IConfiguration configuration;
+        private IGithubClient gitHubClient;
+
+        public StorageWatcher(IHostingEnvironment env, IConfiguration config, IStorageService service)
+        {
+            storageService = service;
+            hostingEnvironment = env;
+            configuration = config;
+            gitHubClient = new GithubClient(env, config);
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(CancellationToken cancellationToken)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task CreateOrUpdatePackage(Package pkg)
```

Comment:
Hi Rekha, just thinking about this approach little bit. We are first updating the blob with new dll and then the table storage. Both these are not one atomic operation. 
How should we handle a situation where the blob gets updated but the table storage operation failed due to any reason?

(However most of the update operations are only dll updates and definition updates, but something to think about here)

Reply:
I have few ideas here, may be we can chat offline about this